spec/lib/diaspora_federation/entities/account_migration_signer_spec.rb

```diff
@@ -0,0 +1,41 @@
+module DiasporaFederation
+  describe Entities::AccountMigrationSigner do
+    let(:entity) { TestAMSignableEntity.new({}) }
+
+    class TestAMSignableEntity < Entity
+      include Entities::AccountMigrationSigner
+
+      property :my_signature, :string, default: nil
+
+      def old_user_id
+        "1"
+      end
+
+      def new_user_id
+        "2"
+      end
+
+      def freeze; end
+    end
+
+    it_behaves_like "a signable" do
+      let(:test_class) { TestAMSignableEntity }
+      let(:test_string) { "AccountMigration:1:2" }
+    end
+
+    describe "#unique_migration_descriptor" do
+      it "composes a string using #old_user_id and #new_user_id" do
+        expect(entity).to receive(:old_user_id).and_return("old")
```

Comment:
I don't think this needs mocking here. I would just use "old" and "new" in the TestAMSignableEntity above and then test that this test and the test below both return "AccountMigration:old:new". I don't like when tests just are an exact copy of the method but with mocking every single method call. I did this myself in the past, but try now not do do it. The problem with this kind of tests is, that you need to change the tests everytime you change something at the code (because they are an exact copy). For example the `#signature_data` test below, you would need to change it, when you rename this `#unique_migration_descriptor` method, even this doesn't change anything with the logic.

So these line by line, method call by method call tests are more complex, take more time to write (you just write the same code again), and make refactorings harder. So I think a little bit less is more in this case.